src/main/scala/ai/tripl/arc/transform/SimilarityJoinTransform.scala

```diff
@@ -0,0 +1,214 @@
+          numPartitions=numPartitions,
+          params=params
+        )
+
+        stage.stageDetail.put("leftView", leftView)  
+        stage.stageDetail.put("leftFields", leftFields.asJava)  
+        stage.stageDetail.put("rightView", rightView)  
+        stage.stageDetail.put("rightFields", rightFields.asJava)  
+        stage.stageDetail.put("outputView", outputView)  
+        stage.stageDetail.put("persist", java.lang.Boolean.valueOf(persist))
+        stage.stageDetail.put("shingleLength", java.lang.Integer.valueOf(shingleLength))
+        stage.stageDetail.put("numHashTables", java.lang.Integer.valueOf(numHashTables)) 
+        stage.stageDetail.put("threshold", java.lang.Double.valueOf(threshold))
+        stage.stageDetail.put("caseSensitive", java.lang.Boolean.valueOf(caseSensitive))
+        stage.stageDetail.put("partitionBy", partitionBy.asJava)
+
+        Right(stage)
+      case _ =>
+        val allErrors: Errors = List(name, description, leftView, leftFields, rightView, rightFields, outputView, persist, shingleLength, numHashTables, threshold, caseSensitive, partitionBy, numPartitions, invalidKeys).collect{ case Left(errs) => errs }.flatten
+        val stageName = stringOrDefault(name, "unnamed stage")
+        val err = StageError(index, stageName, c.origin.lineNumber, allErrors)
+        Left(err :: Nil)
+    }
+  }
+}
+
+case class SimilarityJoinTransformStage(
+    plugin: SimilarityJoinTransform,
+    name: String, 
+    description: Option[String], 
+    leftView: String, 
+    leftFields: List[String], 
+    rightView: String, 
+    rightFields: List[String],
+    outputView: String,
+    persist: Boolean,
+    shingleLength: Int,
+    numHashTables: Int,
+    threshold: Double,
+    caseSensitive: Boolean,
+    partitionBy: List[String], 
+    numPartitions: Option[Int], 
+    params: Map[String, String]
+  ) extends PipelineStage {
+
+  override def execute()(implicit spark: SparkSession, logger: ai.tripl.arc.util.log.logger.Logger, arcContext: ARCContext): Option[DataFrame] = {
+    SimilarityJoinTransformStage.execute(this)
+  }
+}
+
+object SimilarityJoinTransformStage {
+
+  def execute(stage: SimilarityJoinTransformStage)(implicit spark: SparkSession, logger: ai.tripl.arc.util.log.logger.Logger, arcContext: ARCContext): Option[DataFrame] = {
+
+    val uuid = UUID.randomUUID.toString
+
+    // split input string into individual characters
+    val regexTokenizer = { new RegexTokenizer()
+      .setInputCol(uuid)
+      .setPattern("")
+      .setMinTokenLength(1)      
+      .setToLowercase(!stage.caseSensitive)
+    }
+
+    // produce ngrams to group the characters
+    val nGram = { new NGram()
+      .setInputCol(regexTokenizer.getOutputCol)
+      .setN(stage.shingleLength)
+    }
+
+    // convert to vector
+    val countVectorizer = { new CountVectorizer()
+      .setInputCol(nGram.getOutputCol)
+    }
+
+    // build locality-sensitive hashing model
+    val minHashLSH = { new MinHashLSH()
```

Comment:
Minor style suggestion to remove the surrounding brackets as it doesn't require a block (same for above)